Features/original application data #159944560

callaghanc opened the pull request:
This PR is ready for initial review.

Note:
1) Some of the mapping and service work in this PR duplicates some of the mapping and service work we already have. This is intentional! This is part of our journey to move mapping fully into Rails and fully into the new system. Since we don't want to do one giant refactor of the entire app at once, we're doing it piece by piece. Hence, there will be some duplication until we have completed the refactor across the entire app.
2) This PR still needs new tests written and needs to have existing tests pass. @fddayan is working on writing new tests.

fddayan committed:
[#159782560] Introducing CustomApi::ApplicationService

fddayan committed:
wip

fddayan committed:
WIP

fddayan committed:
WIP

callaghanc committed:
[#159944560] Takes model field mapping in a simpler direction

Due to the amount of refactor we are already doing to implement the snapshot application page, we will keep model field mapping simple for now in order to keep this PR understandable by the rest of the team, with the understanding that we can make the mapping more sophisticated in the future.

callaghanc committed:
[#159944560] Adds some custom field mapping behavior for preference model

fddayan committed:
WIP

callaghanc committed:
[#159944560] Continues implementing models and services for fetching snapshot app data

callaghanc committed:
[#159944560]  Adds application members to application for show page

callaghanc committed:
[#159944560] Begins to add preferences to the application on the show page

callaghanc committed:
[#159944560] Adds preferences and attachments to application for show page

callaghanc committed:
[#159944560] Adds flagged applications to the application on the show page

fddayan committed:
[#159944560] Fixed conflicts

fddayan committed:
Merge branch 'master' into features/original-application-data-#159944560

fddayan committed:
WIP

fddayan committed:
WIP

callaghanc committed:
Merge branch 'master' into features/original-application-data-#159944560

akegan commented:
looks good so far to me based on what catherine shared, still missing the page tab piece

akegan commented:
looks like part of the test errors on rake spec are "there's an api request that VCR doesn't know how to handle"

callaghanc commented:
@akegan i'll look at that with @fddayan, we're working on the rspec tests, both new and existing

callaghanc committed:
[#159944560] Adds tabs and breadcrumbs to snapshot page

Also removes unneeded flagged applications from the application object on the supplemental application page.

fddayan committed:
Merge branch 'features/original-application-data-#159944560' of github.com:Exygy/sf-dahlia-lap into features/original-application-data-#159944560

callaghanc committed:
[#159944560] Does a tiny bit of cleanup

callaghanc commented on app/controllers/api/v1/applications_controller.rb:
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
These `# frozen_string_literal: true` comments are required by Rubocop in Ruby 2.3.

akegan replied:
the confusing thing to me is why it isn't causing rubocop to fail when we run it on semaphore, because when I run rubocommit locally it gets made about all the files, not just the files i've touched.

fddayan committed:
WIP

callaghanc commented:
This PR now contains the breadcrumbs and tabs on the application snapshot page.

fddayan committed:
Merge branch 'features/original-application-data-#159944560' of github.com:Exygy/sf-dahlia-lap into features/original-application-data-#159944560

callaghanc committed:
[#159944560] Adds person who claimed name onto preferences

callaghanc committed:
Merge branch 'features/original-application-data-#159944560' of https://github.com/Exygy/sf-dahlia-lap into features/original-application-data-#159944560

callaghanc committed:
[#159944560] Adds rspec tests for application model

callaghanc committed:
[#159944560] Updates unit test for application show page

callaghanc committed:
Merge branch 'master' into features/original-application-data-#159944560

akegan commented on app/controllers/applications_controller.rb:
```diff
@@ -56,7 +58,13 @@ def find_application(id)
     is_listing_lease_up = listing.Status == 'Lease Up'
     application = custom_api_application_service.application(id, snapshot: is_listing_lease_up)
 
+    # Add a flag on the application indicating whether or not it is a snapshot
```
good idea

callaghanc committed:
[#159944560] Updates target Ruby version for Rubocop

callaghanc committed:
Merge branch 'features/original-application-data-#159944560' of https://github.com/Exygy/sf-dahlia-lap into features/original-application-data-#159944560

akegan commented on app/javascript/components/applications/ApplicationPage.js:
```diff
@@ -18,20 +18,44 @@ const buildActionLinkIfNecessary = (app, showAddBtn) => {
+      ]
+    }
+
+    tabSection = {
```
so this means that if you navigate to application through the application tab at the very top, and open an application that's in lease-up, you'll get the breadcrumbs and the tabs, even if you didnt start off from the lease up section, right?

callaghanc replied:
yes

akegan commented on app/models/force/application.rb:
```diff
@@ -17,7 +19,7 @@ class Application < Force::ObjectBase
       { custom_api: 'applicationSubmittedDate', domain: 'application_submitted_date', salesforce: 'Application_Submitted_Date' },
       { custom_api: 'appRTType', domain: '', salesforce: '' },
-      { custom_api: '', domain: 'created_by', salesforce: 'CreatedBy' },
+      { custom_api: '', domain: 'createdby', salesforce: 'CreatedBy' },
```
whoops

callaghanc replied:
this is intentional - the React app expects "createdby". i'd like to change that eventually, but for now the approach in this PR is not to go into the React app and start changing field names

akegan replied:
oops, I didn't mean that it was a mistake that you made that change, more just a mistake that we are expecting createdby in react

akegan commented on spec/javascript/components/applications/__snapshots__/ApplicationPage.test.js.snap:
```diff
@@ -183,7 +183,7 @@ Array [
                   </h4>
                   <p>
-                    Prod Vertiba
+                    None
```
not sure if we want this change

callaghanc replied:
this bit should go away with my next push

akegan commented on spec/javascript/components/applications/__snapshots__/ApplicationPage.test.js.snap:
```diff
@@ -135,7 +135,7 @@ Array [
                   </h4>
                   <p>
-                    1
+                    2
```
was this incorrect before? or unintentional

callaghanc replied:
it was incorrect before

akegan commented on spec/vcr/applications_controller/show/lease_up_application.yml:
```diff
@@ -0,0 +1,466 @@
+        FRANCISCO","mailingAddress":"25 VAN NESS AVE","lastName":"Nerr","languageOther":null,"language":null,"hiv":false,"hasSameAddressAsApplicant":null,"hasAltMailingAddress":false,"genderOther":null,"gender":null,"firstName":"Partha","ethnicity":null,"email":null,"DOB":"1975-03-03","contactId":"0030U00000P8m5oQAB","city":"SAN
+        FRANCISCO","candidateScore":100.00,"appMemberType":"Primary Applicant","appMemberId":"a0n0U0000030tCLQAY","applicationId":"a0o0U000000VVwXQAW","alternatePhoneType":null,"alternatePhone":null,"alternateContactTypeOther":null,"alternateContactType":null,"agency":null,"address":"25
+        VAN NESS AVE"},"otherHousingCounselingAgency":null,"numberOfDependents":null,"name":"APP-00262258","monthlyIncome":null,"lotteryNumberManual":null,"lotteryNumber":"00262258","listingID":"a0W0U000000MX4vUAG","interviewScheduledDate":null,"id":"a0o0U000000VVwXQAW","housingCounselingAgency":null,"householdVouchersSubsidies":"false","householdMembers":[],"householdAssets":null,"HHTotalIncomeW ...
+        555-7777","noPhone":false,"noEmail":false,"noAddress":false,"nickname":null,"middleName":null,"maritalStatus":null,"mailingZip":null,"mailingState":null,"mailingCity":null,"mailingAddress":null,"lastName":"Matts","languageOther":null,"language":null,"hiv":false,"hasSameAddressAsApplicant":null,"hasAltMailingAddress":false,"genderOther":null,"gender":null,"firstName":"Flora","ethnicity":nu ...
```
is this a fake app?

callaghanc replied:
yes, this is test data entered by msol

callaghanc commented:
Update on status of tests in this PR: rspec and unit tests have been added/updated. Am currently working on updating e2e tests.

akegan commented on spec/vcr/applications_controller/show/non_lease_up_application.yml:
```diff
@@ -0,0 +1,584 @@
+        Website","processingStatus":null,"primaryApplicant":{"zip":"94114-2535","yCoordinate":"4545683.891332","xCoordinate":"-13629293.1193475","workInSf":"true","whichComponentOfLocatorWasUsed":"eas","webAppID":"1","state":"CA","sexualOrientationOther":"Ziggy
+        Stardust","sexualOrientation":"Not listed","sexAtBirth":null,"relationship":null,"race":"Other/Multiracial","preferenceStatus":null,"preferenceAddressMatch":"Not
+        Matched","phoneType":"Home","phone":"2222222222","noPhone":false,"noEmail":false,"noAddress":false,"nickname":null,"middleName":null,"maritalStatus":null,"mailingZip":"94114-2535","mailingState":"CA","mailingCity":"SAN
+        FRANCISCO","mailingAddress":"4053 18TH ST","lastName":"Doe","languageOther":null,"language":null,"hiv":false,"hasSameAddressAsApplicant":null,"hasAltMailingAddress":false,"genderOther":"Dothraki","gender":"Not
```
this one looks more like a fake app

callaghanc replied:
this is an auto-generated app for the Automated Test Listing

akegan reviewed (commented):
Mostly just a few questions!

callaghanc committed:
[#159944560] Updates formatting of created by on domain application

callaghanc committed:
Merge branch 'master' into features/original-application-data-#159944560

akegan reviewed (approved):
Approved! hopefully just getting rubocop to shutup now

callaghanc committed:
[#159944560] Updates e2e new application test to use non-lease-up listing

callaghanc commented:
@akegan All tests updated and Semaphore passing.

fddayan reviewed (approved):
looks good 👍

callaghanc committed:
[#159944560] Corrects some preference and app member field mappings

callaghanc committed:
[#159944560] Updates rspec tests

callaghanc committed:
[#159944560] Removes console log
